pkgs/os-specific/darwin/apple-sdk-11.0/frameworks.nix

```diff
@@ -1,4 +1,4 @@
-{ frameworks, libs, libobjc, Libsystem, libnetwork }: with frameworks; with libs;
```

Comment:
@stephank, what exactly do we achieve by dropping `Libsystem` here? I.e., why can't we pass it explicitly to `IOSurface`?

Reply:
This is just an unused parameter, and actually something @tjni added to the commit, I think. I didn't notice it.

As for why not depend on libsystem (instead of xpc), I didn't see any use for something depending on libsystem at all? Isn't it already part of stdenv? I haven't seen any problems leaving it out, but also have not been building much beyond swift and mpv.

Reply:
Ok, same question to @tjni then : )

Should we not be explicit in our Libsystem dependencies? I know stdenv brings it along implicitly but if being explicit doesn't hurt it makes things a lot easier to understand.

(Also, if you changed the commits, you as author seems appropriate, maybe add `Co-authored-by:` if they're still based on stephank's work though.)

Reply:
I also prefer the explicitness, but we actually cannot add the `Libsystem` dependency, because `cc-wrapper` adds the C++ standard library after these dependencies, and both `libc++` and `libstdc++` require `libc` to come after them. I wrote a little bit more about this in https://github.com/NixOS/nixpkgs/pull/185569#issuecomment-1234959249.

Good idea about keeping the authorship intact. I'll try to do this and move the `Libsystem` parameter cleanup into its own commit.

Reply:
Maybe `libstdc++` and `libc++` should be prefixed rather than suffixed? If that doesn't mess anything up but allows us to be explicit about things that'd be a net benefit I think.

If you think the interactions of this would be too hard to foresee then I'm fine with the current state.

Reply:
The am not sure it's the correct thing to do because of the following lookup order:

> The lookup order is as follows:
>
>1. For the quote form of the include directive, the directory of the current file is searched first.
>2. For the quote form of the include directive, the directories specified by -iquote options are searched in left-to-right >order, as they appear on the command line.
>3. Directories specified with -I options are scanned in left-to-right order.
>4. Directories specified with -isystem options are scanned in left-to-right order.
>5. Standard system directories are scanned.
>6. Directories specified with -idirafter options are scanned in left-to-right order.

We add dependency headers using `-isystem`, presumably so that any warnings in dependencies are ignored when compiling the current project. Meanwhile, many a non-nixpkgs build would pick up `lib[std]c++` and `libc` from the standard system directories. We emulate this in nixpkgs by passing `lib[std]c++` and `libc` after other dependencies.

Without knowing any more details, I presume the lookup order was selected intentionally so that we should keep to the current behavior in `cc-wrapper`.

Reply:
I feel like this is valuable reverse engineering, even if it's in part speculative. Do you think adding this as a comment in an appropriate place would allow the next person looking into it to have an easier time or refute the reasoning and be motivated to provide a better explanation?

Reply:
I believe `-idirafter` also interacts with some uses of `#include_next`, where a header 'shadows' another for some reason. I think I saw this somewhere in debugging an error in the Swift build once, but don't remember what it was.

Does anyone know why IOSurface is special at all? Why does it need the libsystem dependency?

Reply:
I think it needs XPC, but because Libsystem also provides XPC two versions ended up in things. The first attempted fix was to have it depend on Libsystem instead but that still lead to two different versions.

Reply:
@toonn I added a commit with a comment including more context to `cc-wrapper`. It is a good suggestion. Thank you.